src/ray/raylet/worker_pool.cc

```diff
@@ -777,6 +784,17 @@ void WorkerPool::OnWorkerStarted(const std::shared_ptr<WorkerInterface> &worker)
   }
 }
 
+void WorkerPool::ExecuteOnPrestartWorkersStarted(std::function<void()> callback) {
+  if (first_job_registered_ || first_job_registered_python_worker_count_ >=
```

Comment:
Should we delete the logic within `OnWorkerStarted`? Seems like a duplicated logic

Reply:
OnWorkerStarted logic is to catch the race between the workers being started with raylet prestarting the workers and the driver connecting to it (which happens in parallel with ray client), the logic in OnWorkerStarted is still needed to have the driver block on worker prestart